lib/wallaroo/core/sink/sink_message_processor.pony

```diff
@@ -62,13 +64,14 @@ class NormalSinkMessageProcessor is SinkMessageProcessor
     sink = s
 
   fun ref process_message[D: Any val](metric_name: String,
-    pipeline_time_spent: U64, data: D, key: Key, i_producer_id: RoutingId,
-    i_producer: Producer, msg_uid: MsgId, frac_ids: FractionalMessageId,
-    i_seq_id: SeqId, latest_ts: U64, metrics_id: U16, worker_ingress_ts: U64)
+    pipeline_time_spent: U64, data: D, key: Key, event_ts: U64,
```

Comment:
Just out of curiosity: is there any way to make this signature more compact or abstract? The exact same code gets repeated in several places, but most of the arguments just get passed on without being examined.

Reply:
This brings us to the same old problem regarding these behavior and method signatures in the hot path.  Ideally, we'd come up with a solution that reduces the noise here, but given results we've seen for attempts in the past, it's something we need to carefully performance test, and hence treat as its own project.